packages/airnode-adapter/src/response-processing/array-type.ts

```diff
@@ -0,0 +1,54 @@
+import { ResponseType, BaseResponseType, baseResponseTypes } from '../types';
+
+// Numeric types could be multiplied by the "_times" reserved parameter
+export function isNumericType(type: ResponseType): type is 'uint256' | 'int256' {
+  return type === 'int256' || type === 'uint256';
+}
+
+export interface ParsedArrayType {
+  readonly baseType: BaseResponseType;
+  readonly dimensions: number;
+}
+
+export function parseArrayType(type: ResponseType): ParsedArrayType | null {
+  if (!type.includes('[')) return null;
+
+  const typeMatch = type.match(/^([^\[]+)(.*)$/);
+  if (!typeMatch || !baseResponseTypes.includes(typeMatch[1] as any)) return null;
```

Comment:
Maybe we should lowercase the input just in case:
```suggestion
  if (!typeMatch || !baseResponseTypes.includes(typeMatch[1].toLocaleLowerCase() as any)) return null;
```

Reply:
do we want that? I mean `UiNt256` should throw an error

Reply:
Correct me if I'm wrong but that value is coming from the config.json, right? Since the API provider might not be so familiar with solidity types he could make a simple mistake and put something like "String", "uInt256" or "Uint256" causing the request to fail.

Troubleshooting the Airnode requests is not a trivial task at the moment (maybe with the RRP Explorer will be easier) so  I'm keen to take a more self-healing approach when possible.

Since this is more of a personal preference you can take it or leave as is.

Reply:
This should be the responsibility of validator to reject such config.json

But you raised a very valid point - currently we only check whether the name of the reserved parameters is correct (https://github.com/api3dao/airnode/blob/feature/gcp-state/packages/airnode-validator/templates/0.3/endpoints.json#L41). I've created https://api3dao.atlassian.net/browse/AN-340 for this